### PR TITLE
Rspec for controllers

### PR DESCRIPTION
**Pull request by Yehezkiel01**
1. Install Rspec-rails Gem along with Factory Bot
2. Complete session_controller_spec
3. Complete welcome_controller_spec

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Metrics/BlockLength:
+  ExcludedMethods: ['describe', 'context']
```
Comment (stevenwjy) [inline .rubocop.yml]:
```suggestion
  ExcludedMethods: ['describe', 'context']
  
```

Comment (stevenwjy) [reply .rubocop.yml]: eh wait in this PR, rubocop gem hasn't been added right?

File: spec/controllers/users/sessions_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+require 'rails_helper'
+
+RSpec.describe Users::SessionsController, type: :controller do
+  before { @request.env['devise.mapping'] = Devise.mappings[:user] }
+
+  describe 'GET #new' do
+    context 'for authenticated user' do
+      before do
+        sign_in create(:user)
```
Comment (stevenwjy) [inline spec/controllers/users/sessions_controller_spec.rb:11]: Where is the value of ":user" location, shouldn't there be some .yml file that contains the data?

Comment (stevenwjy) [reply spec/controllers/users/sessions_controller_spec.rb:11]: and i think that there should be some filtering first in the User model? to ensure that there is no creation of user with invalid email format, too short username, etc.

Comment (Yehezkiel01) [reply spec/controllers/users/sessions_controller_spec.rb:11]: They are defined by factory bot under factories folder

**Review by stevenwjy — APPROVED** [review]

**Pull request merged**